### DISCUSSION TITLE
Error handling with mapRelativeErrors in DataFetcherResult removed

### DISCUSSION THREAD

**Discussion by mukundbabu37** [graphql-java/graphql-java#2882] [Q&A]
We've been using graphql error within a datafetcher result to track all the errors which might arise from calling other APIs and DB lookups. In the resolvers, we just return this data fetcher result and we've been setting the mapRelativeErrors to let the framework map the graphql error from downstream as though its the error from the particular resolver itself. Now that the mapRelativeErrors has been eliminated, the graphql errors as returned as-is. That means its missing the locations and path in the error response. I am posting this to look for an alternate cleaner way to throw the graphql error block with the path and locations in the response.

**Comment by bbakerman** [comment]
>  I am posting this to look for an alternate cleaner way to throw the graphql error block with the path and locations in the response.

If you have code control of your data fetchers, you can use `graphql.execution.DataFetcherResult` to put errors back from a  DF rather than adding them directly via `graphql.execution.ExecutionContext#addError(graphql.GraphQLError)`
